source/shared/core_stmt.cpp

```diff
@@ -796,6 +799,13 @@ bool core_sqlsrv_fetch( _Inout_ sqlsrv_stmt* stmt, _In_ SQLSMALLINT fetch_orient
         // move to the record requested.  For absolute records, we use a 0 based offset, so +1 since 
         // SQLFetchScroll uses a 1 based offset, otherwise for relative, just use the fetch_offset provided.
         SQLRETURN r = stmt->current_results->fetch( fetch_orientation, ( fetch_orientation == SQL_FETCH_RELATIVE ) ? fetch_offset : fetch_offset + 1 TSRMLS_CC );
+        
+        // when AE is on, need to store the current row number being fetched so a cursor can be reset back to it's original position when getting stream data
+        // fwd_row_index is used to store the current row index, which is incremented everytime SQLFetchScroll is called with SQLSRV_CURSOR_FORWARD_ONLY
```

Comment:
When AE is *enabled*, will keep track of the numbers of rows being fetched *so far such that* ... back to *its* original position...